backend/models/user.js

```diff
@@ -58,6 +58,25 @@ const userSchema = new mongoose.Schema({
     default: false,
   },
 
+  userRank: {
```

Comment:
UserRank, UserActivity and UserApp should be the individual model.
You can find the collection structure of three collections from reputation graph.

Reply:
Okay, on it

Reply:
I have made the required changes for adding individual model schemas